### DISCUSSION TITLE
Programing language that can be used for reference

### DISCUSSION THREAD

**Discussion by C-BJ** [rivet-lang/rivet#7] [Ideas]
https://github.com/microsoft/verona and https://github.com/microsoft/BosqueLanguage is Microsoft research project
https://github.com/SerenityOS/jakt is a language with similar design to Rivet.

**Comment by C-BJ** [comment]
https://github.com/r0nsha/chili has some very good designs. You can check its readme or enter its Discord Channel to view the dialogue between me and the project manager

**Comment by C-BJ** [comment]
For the functional part, please also refer to Roc https://www.roc-lang.org/

**Comment by StunxFS** [comment]
Thanks for this, I also use Zig as a reference language.
